lib/tapioca/compilers/symbol_table/symbol_generator.rb

```diff
@@ -464,7 +467,46 @@ def compile_method(symbol_name, constant, method)
           method_name = "#{'self.' if constant.singleton_class?}#{method_name}"
           parameters = "(#{parameters})" if parameters != ""
 
-          indented("def #{method_name}#{parameters}; end")
+          signature_str = indented(compile_signature(signature)) if signature
+          [
+            signature_str,
+            indented("def #{method_name}#{parameters}; end"),
+          ].compact.join("\n")
+        end
+
+        TYPE_PARAMETER_MATCHER = /T\.type_parameter\(([[:word:]]+)\)/
+
+        sig { params(signature: T.untyped).returns(String) }
+        def compile_signature(signature)
```

Comment:
This too could be extracted so we can reuse it from the generators or would it conflict with the way we use Parlour in the generators?

Reply:
Indeed, this can be extracted. Btw, a similar logic is also being used inside `sorbet-pry` as well, maybe it is time to extract a `sorbet-toolkit` gem.

Reply:
Good idea, we could also extract the metrics parser and the config parser in such a gem.